### ISSUE TITLE
Pager content overflows off of it

### ISSUE THREAD

**Issue by ernon66** [Sage/carbon#6563]
### Current behaviour

When the content of a Pager component becomes too long ( longer translations, smaller screen ), it overflows outside of the pager and then outside of the screen

The suggested solution to make some controls disappear based on screen size doesn't allow the user to have access to all the functionalities

Screenshot from the storybook :
![image](https://github.com/Sage/carbon/assets/109949759/53ee735b-057c-4431-aa7d-cc235336f6ac)

### Expected behaviour

The entirety of the elements of the pager should be visible and accessible even on smaller screens

### CodeSandbox or Storybook URL

https://carbon.sage.com/?path=/docs/pager--default

### JIRA Ticket (Sage Only)

SBS-83425

### Suggested Solution

A simple solution might be to allow/default to wrapping which would fix the issue even on very small screens, but I'm not sure about design/UX consequences

Or maybe there's a better solution with a mobile first design 

### Carbon Version

latest (125.9.1)

### What browsers are you seeing the problem on?

Chrome

### What Operating System are you seeing the problem on?

MacOS

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by ernon66** [comment]
Hi @Parsium 👋 

FYI on my side we've landed on a quick fix using the current capabilities of the component (removing some controls, changing translations)

A better medium term solution like wrapping would still be nice to have

I think I've heard about some concept being worked on for a completely responsive pager involving buttons instead of text controls but I don't know more

**Comment by Parsium** [comment]
Thanks for the update, @ernon66! 👍🏼 

We do have the ticket **FE-6378** in our backlog for refactoring the `Pager`. However, further design work will be needed before we can move forward.

I'll close this issue for now given this requires further discussion and you have a workaround.

**Comment by harpalsingh** [comment]
Just wanted to add that there has been discussion on the fully responsive side of this and can that just work automatically. vs the options we added in Carbon to remove elements, i.e the design goes into 2 lines, will confirm who will work on this soon.
